database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```diff
@@ -152,10 +152,8 @@ protected void setField(ResultSet resultSet, StructuredRecord.Builder recordBuil
     } else if (o instanceof BigDecimal) {
       recordBuilder.setDecimal(field.getName(), (BigDecimal) o);
     } else if (o instanceof BigInteger) {
-      if (sqlType == Types.BIGINT && (resultSet.getMetaData().isSigned(columnIndex) ||
-          resultSet.getMetaData().getPrecision(columnIndex) < 19)) {
-        //SQL BIGINT type is 64-bit long thus signed should be able to convert to long without losing precisions
-        // or UNSIGNED type is within the scope of signed long
+      Schema schema = field.getSchema();
+      if (schema.getType() == Schema.Type.LONG) {
```

Comment:
it's possible the Schema is a nullable long, which is represented as a union of long and null. So you need to do something like:
```
schema = schema.isNullable() ? schema.getNonNullable() : schema;
if (schema.getType() == Schema.Type.LONG) {
...
```